python/pyspark/sql/pandas/types.py

```diff
@@ -578,29 +578,44 @@ def _converter(
         if isinstance(dt, ArrayType):
             _element_conv = _converter(dt.elementType, _struct_in_pandas, _ndarray_as_list)
 
-            def convert_array(value: Any) -> Any:
-                if value is None:
-                    return None
-                elif isinstance(value, np.ndarray):
-                    element_list = (
-                        [_element_conv(v) for v in value]
-                        if _element_conv is not None
-                        else [v for v in value]
-                    )
-
-                    if _ndarray_as_list:
+            if _ndarray_as_list:
+
+                def convert_array_ndarray_as_list(value: Any) -> Any:
+                    if value is None:
+                        return None
+                    elif isinstance(value, np.ndarray):
                         # In Arrow Python UDF, ArrayType is converted to `np.ndarray`
                         # whereas a list is expected.
-                        return element_list
+                        return (
+                            [_element_conv(v) for v in value]
+                            if _element_conv is not None
```

Comment:
In this case, `_element_conv` can be `_element_conv or (lambda x: x)` outside of function?

Reply:
Adjusted.